External/Plugins/ASCompletion/PluginMain.cs

```diff
@@ -960,6 +963,59 @@ private void ContextChanged()
             SetItemsEnabled(enableItems, ASContext.Context.CanBuild);
         }
         #endregion
+
+        #region Contextual Code Generation Utilities
+
+        void AddRefactorMenus(List<ICompletionListItem> list, ToolStripMenuItem menu)
+        {
+            if (list == null)
+                return;
+
+            ASComplete.NotifyContextChanged();
```

Comment:
I'm not sure, because following the event handler list when `CodeRefactor.PluginMain` dispatches `"ASCompletion.ContextualGenerator"`, it didn't seem like there was any code for updates. The only reason of calling `NotifyContextChanged` is because it calls `ASComplete.OnResolvedContextChanged`, which calls `CodeRefactor.PluginMain.OnResolvedContextChanged` which then calls `CodeRefactor.PluginMain.UpdateMenuItems`, which is what I want before checking for the `Enabled` of each option. 
I will check if it does get validated automatically without the struggles.

However, as you suggested below, using event mechanism will clearly solve this issue.